src/grimp/adaptors/importscanner.py

```diff
@@ -50,7 +50,9 @@ def scan_for_imports(
 
         try:
             imported_objects = self._get_raw_imported_objects(module_contents)
-        except SyntaxError as e:
+        except rust.ParseError as e:
+            # TODO What to do with the rust.ParseError?
+            # TODO The line number and text are part of str(e), but is there a nice way to get them?
```

Comment:
@seddonym I could do with some help to decide what to do here. The problem is that the line number and text are only available as part of `str(e)`, and not as structured fields.

Reply:
Ah I think I found a reasonable solution now, using `pyclass(extends=PyException)`